### PR TITLE
Feature/kb 12 data dump job

### PR DESCRIPTION
**Pull request by parijke**

### CODE REVIEW

File: data/resource-templates/LDS_Distribution.json [hunk 1]
```diff
@@ -2,21 +2,28 @@
     "o:label": "LDS Distribution",
     "o:resource_template_property": [
         {
-            "o:alternate_label": "Inhouds-URL",
-            "o:alternate_comment": "The URL where the distribution can be directly accessed.",
-            "o:is_required": true,
+            "o:alternate_label": "Naam",
```
Comment (MKodde) [inline data/resource-templates/LDS_Distribution.json:5]: Most of the translations seem English, but the `o:alternate_label` is in Dutch? 

Seems inconsistent

File: data/resource-templates/LDS_Distribution.json [hunk 2]
```diff
@@ -2,21 +2,28 @@
...
+            "data_types": [],
             "vocabulary_namespace_uri": "https:\/\/schema.org\/",
             "vocabulary_label": "Schema.org",
-            "local_name": "contentUrl",
-            "label": "contentUrl"
+            "local_name": "name",
+            "label": "name"
+        },
+        {
+            "o:alternate_label": "Beschrijving",
```
Comment (MKodde) [inline data/resource-templates/LDS_Distribution.json:17]: See previous EN vs NL comment

File: src/Application/Job/RecreateDataCatalogsJob.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+use Omeka\Api\Representation\ItemRepresentation;
+use Omeka\Entity\Job;
+use Omeka\Job\AbstractJob;
+use Omeka\Job\Exception\InvalidArgumentException;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
+
+final class RecreateDataCatalogsJob extends AbstractJob
+{
+    // in php 8.1 convert to enum
```
Comment (MKodde) [inline src/Application/Job/RecreateDataCatalogsJob.php:24]: I don't get this comment, what do enums have to do with the logger?

File: data/resource-templates/LDS_Distribution.json (hunk 1 above)
Comment (parijke) [reply data/resource-templates/LDS_Distribution.json:5]: Is the choice of the PO

File: src/Application/Service/CatalogDumpService.php
Comment (MKodde) [inline src/Application/Service/CatalogDumpService.php:1]: May I suggest you update the file to adhere to the chosen (psr-12?) standard. I now see the lower_under naming strategy emerge once again :) I'm guessing Omeka S chose the PSR-12 standard as it's go to coding standard. But I might be wrong. Please ahere to their standard. And actively check it using the PHP CS suite.

~`$publisher_uri`~ `$publisherUri`

PHPCS can probably easily fix this for you using the built in [phpcbf](https://phpqa.io/projects/phpcbf.html)

File: src/Application/Service/CatalogDumpService.php [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        }
+
+        # Step 5 - remove Omeka classes and properties (o:)
+        $this->removeOmekaTags($graph);
+
+        # Step 6 - output the graph in several serializations
+        $this->dumpSerialisedFiles($graph);
+    }
+
+    // Maybe this step isn't needed, Bob said on 11th April.
```
Comment (MKodde) [inline src/Application/Service/CatalogDumpService.php:90]: I'd refrain from adding this kind of info to the actual code base and offload it to the project board. This comment has meaning to you, and a little bit to me, but any contributor outside our organization would be clueless as to what this means.

File: src/Application/Service/ItemSetCrawler.php [hunk 5]
```diff
@@ -16,20 +18,18 @@ final class ItemSetCrawler
     private const SORT_BY = 'id';
     private const SORT_ORDER = 'asc';
 
-    private $resources_seen = [];
-    private $resources_todo = [];
-    private $item_set_id;
+    private $resourcesProcessed = [];
+    private $resourcesToProcess = [];
+    private $itemSetId;
```
Comment (MKodde) [inline src/Application/Service/ItemSetCrawler.php:23]: :heart_decoration:

File: src/Infrastructure/Exception/CatalogResourceTemplateDoesNotExists.php [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LinkedDataSets\Infrastructure\Exception;
```
Comment (MKodde) [inline src/Infrastructure/Exception/CatalogResourceTemplateDoesNotExists.php:5]: Why are these exceptions in the infra namespace? If anything, I'd sooner say they are a part of the domain logic. As you throw them when a domain rule is broken. 

Maybe I do not see the full picture?

File: src/Generic/InstallResources.php
Comment (MKodde) [inline src/Generic/InstallResources.php:1]: Are the InstallResources.php and AbstractModule.php actually used?

If you are (to use) using them consider the following:

1. What is a Generic namespace? I feel these are infrastructural classes. Consider moving them there
2. Given these are classes lifted from another resource, is it possible to install them via Composer? Now we include code from an outside party that we now need to maintain. That might not be what we want?
3. Autoloading these classes will prove difficult as they do not adhere to the PSR4 standard. And I do not see an addition to the composer.json classmap or psr0 section.

**Review by MKodde — CHANGES_REQUESTED** [review]
Hi @parijke, nice work. The code base is definitely moving in the right direction. 

I found it hard to focus my review as this PR is kind of all over the place. 
1. Adding a PR description can help describe what you did an why
2. Breaking up features in smaller PR's would also help in this regard

Hope you benefit from my feedback, and if you want specific attention to a certain piece of this PR, feel free to ask me.

File: data/resource-templates/LDS_Distribution.json (hunk 2 above)
Comment (parijke) [reply data/resource-templates/LDS_Distribution.json:17]: Choice of the PO

File: src/Application/Service/CatalogDumpService.php [hunk 7]
```diff
@@ -2,90 +2,88 @@
...
-        if (!$this->id) {
-            throw new InvalidArgumentException('No catalog_id was provided to the job');
+    public function __construct(
+        LoggerInterface $logger,
+        $uriHelper
+    ) {
+        $this->logger = $logger;
+        $this->uriHelper = $uriHelper;
+        if (!$this->uriHelper) {
+            throw new ServiceNotFoundException('The UriHelper service is not found');
```
Comment (MKodde) [inline src/Application/Service/CatalogDumpService.php:34]: In general, this construction can be covered by requiring a typed version of the helper class in the constructors parameter. Also by not allowing a null value. That way the error would be automatically triggered without you having to add logic to verify correct dependencies are set.

File: src/Infrastructure/Services/Factories/ApiManagerHelperFactory.php [hunk 8]
```diff
@@ -0,0 +1,17 @@
+<?php
```
Comment (MKodde) [inline src/Infrastructure/Services/Factories/ApiManagerHelperFactory.php:1]: May I suggest you require strict type checking throughout the application.

`<?php declare(strict_types=1);`

File: test/bootstrap.php [hunk 9]
```diff
@@ -2,11 +2,13 @@
 
 declare(strict_types=1);
 
-require dirname(dirname(dirname(__DIR__))) . '/bootstrap.php';
+//require dirname(__DIR__, 3). '/bootstrap.php';
```
Comment (MKodde) [inline test/bootstrap.php:5]: Cool to see the static analysis actually work now on github!

1. Maybe try to configure phpstan at a level that satisfies most important things we want enforced in our project, but low enough to not nag about every nit it finds?
2. And maybe if time allows, address some more of the findings PHPStan now comes up with. A green build would be awesome on `main`.

**Review by MKodde — COMMENTED** [review]
Nice cleanup work has been done! Thanks @parijke !

Before we can merge this, the merge conflict should be resolved. A rebase with main should be quite painless (:crossed_fingers:).

See some additional feedback below. As stated IRL, I tried to be pragmatic while reviewing the code. As this is the product of more than one other branches, and the remaining time to spend is limited.

**Review by coret — APPROVED** [review]

**Pull request merged**